packages/x-internals/src/export/loadStyleSheets.tsx

```diff
@@ -0,0 +1,51 @@
+/**
+ * Loads all stylesheets from the given root element into the document.
+ * @returns an array of promises that resolve when each stylesheet is loaded
+ * @param document Document to load stylesheets into
+ * @param root Document or ShadowRoot to load stylesheets from
+ */
+export function loadStyleSheets(document: Document, root: Document | ShadowRoot) {
+  const stylesheetLoadPromises: Promise<void>[] = [];
+  const headStyleElements = root.querySelectorAll("style, link[rel='stylesheet']");
+
+  for (let i = 0; i < headStyleElements.length; i += 1) {
+    const node = headStyleElements[i];
+    if (node.tagName === 'STYLE') {
+      const newHeadStyleElements = document.createElement(node.tagName);
+      const sheet = (node as HTMLStyleElement).sheet;
+
+      if (sheet) {
+        let styleCSS = '';
+        // NOTE: for-of is not supported by IE
```

Comment:
```suggestion
```
Stale comment, we dropped IE support back in v7

Reply:
Yeah, I think I might change this into a `for...of` in a follow-up PR.

Reply:
`for...of` doesn't seem to be any better than a regular `for` loop in this case though

Reply:
Yeah, it would be mostly for ergonomics.